### PR TITLE
[core] Factor timePoint initialization: take 2

### LINKED ISSUE #9140: Investigate transition issues when time point is not current in still mode
Follow-up to #9130.

We should be able to calculate `timePoint` when instantiating it without problems. It could be that we're using this value to trigger style transition updates that we haven't accounted for on the original patch.

/cc @kkaefer @jfirebaugh

### PR DESCRIPTION
**Pull request by brunoabinader**
First attempt at #8880 causes rendering artifacts when using the same map object to render the same region in different zoom levels.

This happens because the [time point sent to ZoomHistory](https://github.com/mapbox/mapbox-gl-native/blob/master/src/mbgl/renderer/render_style.cpp#L86) is always the same in still mode with the new approach, causing cross faded property evaluations to fail, because [its transition delta](https://github.com/mapbox/mapbox-gl-native/blob/master/src/mbgl/renderer/cross_faded_property_evaluator.cpp#L30) is computed using `ZoomHistory::lastIntegerZoomTime` which equals to `parameters.now`, causing `t` to become `0`.

Added a unit test to verify: `API.ZoomHistory`

Fixes #9140.

/cc @bsudekum

### CODE REVIEW

**Comment by brunoabinader** [comment]
CircleCI jobs are failing due to a somewhat unrelated error - the expectation result for `API.RepeatedRender` I've generated locally slightly differs from the one generated via the software renderer inside the VM - I'll update the expectation file from the actual file generated inside the VM accordingly.

File: src/mbgl/map/zoom_history.hpp [hunk 1]
```diff
@@ -13,19 +13,20 @@ struct ZoomHistory {
     bool first = true;
 
     bool update(float z, const TimePoint& now) {
+        static TimePoint zero = TimePoint(Duration::zero());
```
Comment (kkaefer) [inline src/mbgl/map/zoom_history.hpp]: No need to make this static; likely the compiler can replace `TimePoint(Duration::zero());` with an inline 0 value. You can use `constexpr` as an alias instead.

**Comment by brunoabinader** [comment]
> If we used Clock::time_point::zero() instead of Clock::time_point::max(), would that eliminate the need for special cases in ZoomHistory?

Not quite, because [t would still evaluate to 0](https://github.com/mapbox/mapbox-gl-native/blob/master/src/mbgl/renderer/cross_faded_property_evaluator.cpp#L30) in `CrossFadedPropertyEvaluator::evaluate`.

**Comment by brunoabinader** [comment]
Address & Thread sanitizers are now covered in #9399.

**Pull request merged**